content/blog/2020-08-19-train-a-sequence-model-with-poutyne/index.en.md

```diff
@@ -0,0 +1,598 @@
+if you don't have one, you can use the Google Colab notebook). 
+
+Second, we set the batch size (i.e. the number of elements to see before updating the model), the learning rate for the optimizer
+and the number of epochs.
+
+```python
+device = torch.device("cuda:0")
+
+batch_size = 128
+lr = 0.1
+
+epoch_number = 10
+```
+
+We also need to set Pythons's, NumPy's and PyTorch's random seeds using the Poutyne function to make our training (almost) completely reproducible.
+
+> See [here](https://determined.ai/blog/reproducibility-in-ml/) for an explanation on why setting seed does not guarantee complete reproducibility.
+
+```python
+set_seeds(42)
+```
+
+## The Dataset
+The dataset consists of `1,010,987` complete French and English Canadian addresses and their associated tags.
+Here's an example address
+
+`"420 rue des Lilas Ouest, Québec, G1V 2V3"`
+
+and its corresponding tags
+
+`[StreetNumber, StreetName, StreetName, StreetName, Orientation, Municipality, PostalCode, PostalCode]`.
+
+Now let's download our dataset. For simplicity, a `100,000` addresses test set is kept aside, with 80% of the remaining addresses used for training and 20 % used as a validation set. 
+Also note that the dataset was pickled for simplicity (using a Python `list`). Here is the code to download it.
+
+```python
+def download_data(saving_dir, data_type):
+    """
+    Function to download the dataset using data_type to specify if we want the train, valid or test.
+    """
+
+    # hardcoded url to download the pickled dataset
+    root_url = "https://dot-layer.github.io/blog-external-assets/train_rnn/{}.p"
+
+    url = root_url.format(data_type)
+    r = requests.get(url)
+    os.makedirs(saving_dir, exist_ok=True)
+
+    open(os.path.join(saving_dir, f"{data_type}.p"), 'wb').write(r.content)
+
+
+download_data('./data/', "train")
+download_data('./data/', "valid")
+download_data('./data/', "test")
+```
+
+Now let's load the data in memory.
+
+```python
+# load the data
+train_data = pickle.load(open("./data/train.p", "rb"))  # 728,789 examples
+valid_data = pickle.load(open("./data/valid.p", "rb"))  # 182,198 examples
+test_data = pickle.load(open("./data/test.p", "rb"))  # 100,000 examples
+```
+
+As explained before, the (train) dataset is a list of `728,789` tuples where the first element is the full address, and the second is a list of tags (the ground truth).
+
+```python
+train_data[:2] # The first two train items
+```
+
+> (the output)
+
+![data_snapshot](data_snapshot.png)
+
+### Vectorize the Dataset
+
+Since we used word embeddings as the encoded representations of the words in the addresses, we need to *convert* the addresses into the corresponding word vectors. In order to do that, we will use a `vectorizer` (i.e. the process of converting words into vectors). This embedding vectorizer will extract, for each word, the embedding value based on the pre-trained French fastText model.
```

Comment:
All my concerns about embeddings vs corpus languages are solved except here. I would simply specify French embeddings are used because it is the most prominent language in the dataset used.

```suggestion
Since we used word embeddings as the encoded representations of the words in the addresses, we need to *convert* the addresses into the corresponding word vectors. In order to do that, we will use a `vectorizer` (i.e. the process of converting words into vectors). This embedding vectorizer will extract, for each word, the embedding value based on the pre-trained French fastText model. We use French embeddings because French is the language in which most of the adresses in our dataset are written.
```